modules/top-level/plugins/combine-plugins.nix

```diff
@@ -0,0 +1,102 @@
+          configurePhase = ''
+            ${prev.configurePhase or ""}
+            rm -vf doc/tags'';
+        })
+      ) pluginsToCombine;
+
+      # Gather python 3 dependencies from every plugins
+      python3Dependencies =
+        let
+          deps = map (p: p.plugin.python3Dependencies or (_: [ ])) pluginsToCombine;
+        in
+        ps: builtins.concatMap (f: f ps) deps;
+
+      # Combined plugin
+      combinedPlugin = pkgs.vimUtils.toVimPlugin (
+        pkgs.buildEnv {
+          name = "plugin-pack";
+          paths = overriddenPlugins;
+          inherit pathsToLink;
+          # Remove empty directories and activate vimGenDocHook
+          postBuild = ''
+            find $out -type d -empty -delete
+            runHook preFixup
+          '';
+          passthru = {
+            inherit python3Dependencies;
+          };
+        }
+      );
+
+      # Combined plugin configs
+      combinedConfig = builtins.concatStringsSep "\n" (
+        builtins.concatMap (x: lib.optional (x.config != null && x.config != "") x.config) pluginsToCombine
+      );
+    in
+    normalizePlugin {
+      plugin = combinedPlugin;
+      config = combinedConfig;
+    };
+
+  # Take a list of combined plugins, combine the relevant ones and return the resulting list of plugins
+  combinePlugins =
+    {
+      standalonePlugins,
+      pathsToLink,
+      normalizedPlugins,
+    }:
+    let
+      # Plugin list extended with dependencies
+      allPlugins =
+        let
+          pluginWithItsDeps =
+            p: [ p ] ++ builtins.concatMap pluginWithItsDeps (normalizePluginList p.plugin.dependencies or [ ]);
+        in
+        lib.unique (builtins.concatMap pluginWithItsDeps normalizedPlugins);
+
+      # Separated start and opt plugins
+      partitionedOptStartPlugins = builtins.partition (p: p.optional) allPlugins;
+      startPlugins = partitionedOptStartPlugins.wrong;
+      # Remove opt plugin dependencies since they are already available in start plugins
+      optPlugins = __removeDependencies partitionedOptStartPlugins.right;
+
+      # Test if plugin shouldn't be included in plugin pack
+      isStandalone =
+        p:
+        builtins.elem p.plugin standalonePlugins || builtins.elem (lib.getName p.plugin) standalonePlugins;
+
+      # Separated standalone and combined start plugins
+      partitionedStandaloneStartPlugins = builtins.partition isStandalone startPlugins;
+      pluginsToCombine = partitionedStandaloneStartPlugins.wrong;
+      # Remove standalone plugin dependencies since they are already available in start plugins
+      standaloneStartPlugins = __removeDependencies partitionedStandaloneStartPlugins.right;
+
+      # Combine start plugins into a single pack
+      pluginPack = __mkPluginPack { inherit pluginsToCombine pathsToLink; };
+
+      # Combined plugins
+      combinedPlugins = [ pluginPack ] ++ standaloneStartPlugins ++ optPlugins;
+    in
+    combinedPlugins;
+in
+{
+  inherit combinePlugins;
```

Comment:
Why not just return the function itself? Like we do in other single-util-function files (mkLsp, mkCmpPlugin, mkVimPlugin, mkNeovimPlugin, etc).

Returning an attrset implies this file may down-the-line be returning a bunch of util functions, which I think would not be ideal in this case.

I'd also suggest inlining the `combinePlugins` implementation, unless we need a recursive binding? If we do need recursion, we could also use `lib.fix`.